packages/gatsby/cache-dir/app.js

```diff
@@ -8,18 +8,34 @@ import emitter from "./emitter"
 import { apiRunner, apiRunnerAsync } from "./api-runner-browser"
 import { setLoader, publicLoader } from "./loader"
 import DevLoader from "./dev-loader"
-import syncRequires from "$virtual/sync-requires"
 // Generated during bootstrap
 import matchPaths from "$virtual/match-paths.json"
 
 window.___emitter = emitter
 
-const loader = new DevLoader(syncRequires, matchPaths)
+let pageComponentRequires
+if (process.env.GATSBY_EXPERIMENT_LAZY_DEVJS) {
+  pageComponentRequires = require(`$virtual/lazy-client-sync-requires`)
+} else {
+  pageComponentRequires = require(`$virtual/sync-requires`)
+}
+
+const loader = new DevLoader(pageComponentRequires, matchPaths)
 setLoader(loader)
 loader.setApiRunner(apiRunner)
 
 window.___loader = publicLoader
 
+// Do dummy dynamic import so the jsonp __webpack_require__.e is added to the commons.js
+// bundle. This ensures hot reloading doesn't break when someone first adds
+// a dynamic import.
+//
+// Without this, the runtime breaks with a
+// "TypeError: __webpack_require__.e is not a function"
+// error.
+// eslint-disable-next-line
+import("./dummy")
+
```

Comment:
Is it ok to keep this for non `GATSBY_EXPERIMENT_LAZY_DEVJS` runs?

Can it actually help in regular `gatsby develop`? If not maybe this should be just in `if (process.env.GATSBY_EXPERIMENT_LAZY_DEVJS) {` block?

Reply:
yeah this is a general fix so I didn't think it needed to be behind a flag